compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2135,6 +2137,7 @@ declare_lint! {
     /// ```rust,compile_fail
     /// # #![allow(unused)]
     /// #![deny(explicit_outlives_requirements)]
+    /// #![deny(warnings)]
```

Comment:
Why do we need this here? Does this code not trigger `explicit_outlives_requirements` (that would be pretty bad lol)?

Reply:
It triggers it but as a `warn`, so I had to deny warnings in order to fail the compilation of it.

here is the stdout when there is no `#![deny(warnings)]`:
```text
---- src/builtin.rs - builtin::EXPLICIT_OUTLIVES_REQUIREMENTS (line 2136) stdout ----
warning: outlives requirements can be inferred
 --> src/builtin.rs:2140:24
  |
6 |   struct SharedRef<'a, T>
  |  ________________________^
7 | | where
8 | |     T: 'a,
  | |__________^ help: remove this bound
  |
note: the lint level is defined here
 --> <crate attribute>:1:6
  |
1 | warn(rust_2018_idioms)
  |      ^^^^^^^^^^^^^^^^
  = note: `#[warn(explicit_outlives_requirements)]` implied by `#[warn(rust_2018_idioms)]`

warning: 1 warning emitted

Test compiled successfully, but it's marked `compile_fail`.
```

Reply:
ugh, does the warn override the deny :(

Reply:
If it's not a bug, it does override :)